src/algebraic_geometry/morphisms/quasi_compact.lean

```diff
@@ -114,4 +114,53 @@ begin
     exact supr_insert }
 end
 
+lemma exists_pow_mul_eq_zero_of_res_basic_open_eq_zero_of_is_affine_open (X : Scheme)
+  {U : opens X.carrier} (hU : is_affine_open U) (x f : X.presheaf.obj (op U))
+  (H : X.presheaf.map (hom_of_le $ X.basic_open_subset f : X.basic_open f ⟶ U).op x = 0) :
```

Comment:
Can we do something to make this less verbose? We just want to restrict `x` to `D(f)`, right?

Reply:
Using this definiton:
```
local attribute [instance] concrete_category.has_coe_to_sort concrete_category.has_coe_to_fun

@[reducible] def _root_.Top.presheaf.restrict {X : Top} {C : Type*} [category C] [concrete_category C]
  {F : X.presheaf C} {U V : opens X} (x : F.obj (op V)) (h : U ≤ V) : F.obj (op U) :=
F.map (hom_of_le h).op x

local infix ` ∣_ₛ `: 80 := _root_.Top.presheaf.restrict
```

This can be shortened to `x ∣_ₛ X.basic_open_subset f`.
I can open a separate PR implementing this looks better?

Reply:
I think that would certainly make this statement more readable!